src/coreComponents/python/modules/geosx_mesh_doctor/checks/vtk_polyhedron.py

```diff
@@ -0,0 +1,202 @@
+
+    def dump(self) -> Sequence[int]:
+        """
+        Returns the face stream awaited by vtk, but in a python container.
+        The content can be used, once converted to a vtkIdList, to define another polyhedron in vtk.
+        :return: The face stream in a python container.
+        """
+        result = [len(self.__data)]
+        for face_nodes in self.__data:
+            result.append(len(face_nodes))
+            result += face_nodes
+        return tuple(result)
+
+    def __repr__(self):
+        result = [str(len(self.__data))]
+        for face_nodes in self.__data:
+            result.append(str(len(face_nodes)))
+            result.append(", ".join(map(str, face_nodes)))
+        return ",\n".join(result)
+
+
+def build_cell_graph(face_stream: FaceStream, add_compatibility=False) -> networkx.Graph:
+    """
+    Given a face stream/polyhedron, builds the connections between the faces.
+    Those connections happen when two faces share an edge.
+    :param face_stream: The face stream description of the polyhedron.
+    :param add_compatibility: Two faces are considered compatible if their normals point in the same direction (inwards or outwards).
+        If `add_compatibility=True`, we add a `compatible={"-", "+"}` flag on the edges
+        to indicate that the two connected faces are compatible or not.
+        If `add_compatibility=False`, non-compatible faces are simply not connected by any edge.
+    :return: A graph which nodes are actually the faces of the polyhedron.
+        Two nodes of the graph are connected if they share an edge.
+    """
+    edges_to_face_indices = defaultdict(list)
+    for face_index, face_nodes in enumerate(face_stream.face_nodes):
+        # Each edge is defined by two nodes. We do a small trick to loop on consecutive points.
+        for face_indices in zip(face_nodes, face_nodes[1:] + (face_nodes[0], )):
+            edges_to_face_indices[frozenset(face_indices)].append(face_index)
+    # We are doing here some small validations w.r.t. the connections of the faces
+    # which may only make sense in the context of numerical simulations.
+    # As such, an error will be thrown in case the polyhedron is not closed.
+    # So there may be a lack of absolute genericity, and the code may evolve if needed.
+    for face_indices in edges_to_face_indices.values():
+        assert len(face_indices) == 2
+    # Computing the graph degree for validation
+    degrees = defaultdict(int)
+    for face_indices in edges_to_face_indices.values():
+        for face_index in face_indices:
+            degrees[face_index] += 1
+    for face_index, degree in degrees.items():
+        assert len(face_stream[face_index]) == degree
+    # Validation that there is one unique edge connecting two faces.
+    face_indices_to_edge_index = defaultdict(list)
+    for edge_index, face_indices in edges_to_face_indices.items():
+        face_indices_to_edge_index[frozenset(face_indices)].append(edge_index)
```

Comment:
- The face indices are meant to be unique, so I use a `set` and not a simple iterable.
- A `set` is mutable, so it cannot be used a key in a dictionary, since the key could be modified after its insertion in the mapping (there's also some hashing involved). `frozenset` is immutable so it removes this issue.